src/uproot/writing/_cascadetree.py

```diff
@@ -129,22 +128,18 @@ def __init__(
                 except TypeError:
                     try:
                         awkward = uproot.extras.awkward()
-                    except ImportError as err:
+                    except ModuleNotFoundError as err:
                         raise TypeError(
-                            "not a NumPy dtype and 'awkward' cannot be imported: {0}\n\n{1}".format(
-                                repr(branch_type), str(err)
-                            )
-                        )
+                            f"not a NumPy dtype and 'awkward' cannot be imported: {branch_type!r}"
+                        ) from err
```

Comment:
Here's the biggest manual change. Since Python 3 allows exception chaining (in fact, it's the default, though I made it explicit here), the original exception is already present, so I removed appending it manually. Also this has always been the wrong exception, you should not manually throw a ImportError, but it's subclass, ModuleNotFoundError, in Python 3.

Reply:
Okay. And it can be ModuleNotFoundError now. (I'm pretty sure that word didn't exist in Python 2, which is why we were catching the superclass.)

Reply:
Yes, again, all my statements are based on Python 3, not saying this should have been different in Python 2. Though this was always wrong in Python 3, you should be able to catch a ModuleNotFound, ImportError can also mean other import-related failures when a module is present but broken.

Reply:
But it would have been irritating to try to toggle based on Python version.